openraft/src/engine/initialize_test.rs

```diff
@@ -0,0 +1,120 @@
+use maplit::btreeset;
+
+use crate::engine::Command;
+use crate::engine::Engine;
+use crate::entry::EntryRef;
+use crate::error::InitializeError;
+use crate::error::NotAMembershipEntry;
+use crate::error::NotAllowed;
+use crate::error::NotInMembers;
+use crate::EntryPayload;
+use crate::LeaderId;
+use crate::LogId;
+use crate::Membership;
+use crate::MetricsChangeFlags;
+use crate::Vote;
+
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub(crate) struct Req {}
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub(crate) struct Resp {}
+
+crate::declare_raft_types!(
+   pub(crate) Config: D = Req, R = Resp, NodeId = u64
+);
+
+#[test]
+fn test_initialize() -> anyhow::Result<()> {
+    let eng = Engine::<u64>::default;
+
+    let log_id0 = LogId {
+        leader_id: LeaderId::new(0, 0),
+        index: 0,
+    };
+    let vote0 = Vote::new(0, 0);
+
+    let m12 = || Membership::<u64>::new(vec![btreeset! {1,2}], None);
+    let payload = EntryPayload::<Config>::Membership(m12());
+    let mut entries = [EntryRef::new(&payload)];
+
```

Comment:
could add more comments in each test case?

such as the first case is `test add node 1,2 into cluster`